grc/core/blocks/options.py

```diff
@@ -0,0 +1,362 @@
+        Args:
+            grc_parameters (str): option block parameter values from grc
+        """
+        self.params['output_language'].value = grc_parameters.get('output_language')
+        self.params['generate_options'].value = grc_parameters.get('generate_options')
+        self.update_current_workflow()
+
+        # fill default values of each workflow parameters from grc_parameters
+        for i in range(len(self.workflow_params)):
+            self.workflow_params[i]['params']['id'].value = grc_parameters.get('id')
+            for default_id in self.default_ids:
+                # the value of output_language and generate_options already been defined in parse_workflows
+                if default_id == 'output_language' or default_id == 'generate_options':
+                    continue
+                self.workflow_params[i]['params'][default_id].value = grc_parameters.get(default_id)
+
+        self.params = self.get_params_by_workflow_id(self.current_workflow.id)
+        self.parent.validate()  # validiate the flow graph
+
+    def rewrite(self) -> None:
+        """
+        Get parameters based on current workflow
+        Update generated_options each time the value of output_language is changed
+        Update generator class based on current workflow
+        """
+        self.update_current_workflow()
+        # Make a copy of the current parameter values
+        backup_params = {
+            k: self.params[k].value
+            for k in ['title', 'author', 'copyright', 'description', 'output_language', 'generate_options']
+        }
+        # Create new params dictionary based on current workflow
+        self.params = self.get_params_by_workflow_id(self.current_workflow.id)
+        # Re-assign old values
+        for k, v in backup_params.items():
+            self.params[k].value = v
+
+        # re assign options of generate_options based on current output language
+        chosen_language = self.params['output_language'].value
+        output_language_pair = ()
+        for output_language, output_language_label in self.codegen_options.keys():
+            if chosen_language == output_language:
+                output_language_pair = (output_language, output_language_label)
+                break
+
+        list_of_generated_options_pair = self.codegen_options[output_language_pair]
+
+        tmp_dct = OrderedDict()
+        tmp_dct.attributes = defaultdict(dict)
+        for generated_options, generated_options_label in list_of_generated_options_pair:
+            tmp_dct[generated_options] = generated_options_label
+        self.params['generate_options'].options = tmp_dct
+
+        # Update the generator_class_name and generator_module parameters based
+        # on the current workflow
+        self.params['generator_class_name'].set_value(self.current_workflow.generator_class)
+        self.params['generator_module'].set_value(self.current_workflow.generator_module)
+        Element.rewrite(self)
```

Comment:
superfluous?

Reply:
Yes, good catch. I was debating whether or not to remove this line or the `super().rewrite()`, because we don't use the super-rewrite anywhere else. That said, this does not inherit directly from `Element` so I removed this line like you suggested.